sdk/storage/inc/datalake/datalake_utilities.hpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "common/crypt.hpp"
+
+#include <map>
+#include <string>
+
+namespace Azure { namespace Storage { namespace DataLake {
+
+  inline std::map<std::string, std::string> ParseDataLakePropertiesString(
+      const std::string& dataLakePropertiesString)
```

Comment:
This whole function is kind of messy. Can you refactor it?

Reply:
ref https://github.com/Azure/azure-sdk-for-cpp/blob/master/sdk/storage/src/common/storage_credential.cpp

Reply:
True, it's some midnight mess-around with no testing and validation done. I'll rewrite the whole thing.